src/Microsoft.Health.Fhir.Core/Extensions/IntersectExtensions.cs

```diff
@@ -71,7 +76,7 @@ internal static class IntersectExtensions
                 issues.Add(string.Format(CultureInfo.InvariantCulture, Core.Resources.InvalidListConfigSetting, fieldName));
             }
 
-            if (config.Select(selector).GroupBy(x => x).Any(x => x.Count() > 1))
+            if (config.Select(selectorDistinct).GroupBy(x => x).Any(x => x.Count() > 1))
```

Comment:
I ended up reverting this change from my pr. 
It really depends if having the same search parameter listed with different 'profiles' is valid or not? I know it used to break api management